File: src/ImageViewer/Models/ImageInfo.cs
namespace ImageViewer.Models;

public class ImageInfo
{
    public string FilePath { get; set; } = string.Empty;
    public string FileName { get; set; } = string.Empty;
    public string FileSize { get; set; } = string.Empty;
    public DateTime DateModified { get; set; }
    public string Extension { get; set; } = string.Empty;
}

File: src/ImageViewer-mac/ImageViewer-mac/Models/ImageInfo.swift
//
//  ImageInfo.swift
//  ImageViewer-mac
//
//  Created by Michael Hammond on 7/16/25.
//

import Foundation

struct ImageInfo: Identifiable, Hashable {
    let id = UUID()
    let filePath: String
    let fileName: String
    let fileSize: String
    let dateModified: Date
    let fileExtension: String
    
    var isAnimatedGif: Bool {
        fileExtension.lowercased() == ".gif"
    }
    
    var isVideo: Bool {
        fileExtension.lowercased() == ".mp4"
    }
    
    var isStaticImage: Bool {
        !isAnimatedGif && !isVideo
    }
    
    var hasLoopableContent: Bool {
        isAnimatedGif || isVideo
    }
}